lib/facility.rb

```diff
@@ -1,14 +1,51 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :dmv_name, 
+              :dmv_address, 
+              :dmv_phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @dmv_name = ([:office_name] || [:dmv_office]) || [:name]
+    @dmv_address = ([:street_address_line_1] || [:address_li]) || [:address1]
+    @dmv_phone = [:public_phone_number] ||[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @fees = {antique: 25, ev: 200, regular: 100} 
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  
+  def register_vehicle(vehicle) 
+    if @services.include?('Vehicle Registration')
+      vehicle.update_registration_date
+      vehicle.assign_plate 
+      @collected_fees += @fees[vehicle.plate_type]
+      @registered_vehicles << vehicle
+    else
+      "Denied"
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') && (registrant.permit && registrant.age >=16)
+        registrant.license_data[:written] = true
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') && registrant.license_data[:written]
+      registrant.license_data[:license] = true
+    end
+  end
+
+  def renew_drivers_license(registrant)
```

Comment:
Where are all you tests for these methods? I didn't see them in the facility_spec.rb file. Remember you want to write tests for every method you write. These ones have a lot of conditions that need to be met so we need to write tests that test every condition combination.